addons/sourcemod/scripting/vsh/abilities/ability_force_forward.sp

```diff
@@ -96,4 +124,28 @@ methodmap CForceForward < SaxtonHaleBase
 			g_flSpeedRageBonusMultiplier[this.iClient] = this.flSpeedRageMultValue;
 		}
 	}
-};
\ No newline at end of file
+	
+	public void OnEntityCreated(int iEntity, const char[] sClassname)
+	{
+		if (StrEqual(sClassname, "trigger_catapult") || StrEqual(sClassname, "trigger_push"))
+		{
+			SDKHook(iEntity, SDKHook_StartTouch, OnCatapultStart);
+			SDKHook(iEntity, SDKHook_EndTouch, OnCatapultEnd);
+		}
+	}
+};
+
+public Action OnCatapultStart(int iEntity, int iClient)
+{
+	g_bInCatapult[iClient] = true;
```

Comment:
Another one, i believe `iClient`, the toucher is not always client, can be any other entity so a check would be needed